gdal/frmts/hdf5/hdf5dataset.cpp

```diff
@@ -79,6 +79,7 @@ void GDALRegister_HDF5()
 
     GDALDriver *poDriver = new GDALDriver();
 
+    poDriver->SetMetadataItem(GDAL_DCAP_VIRTUALIO, "YES");
```

Comment:
to protect with #if H5_VERS_MAJOR == 1 && H5_VERS_MINOR <= 8 && (H5_VERS_MINOR < 8 || H5_VERS_RELEASE < 9)

As it is used twice, I'd suggest a
```
#if H5_VERS_MAJOR == 1 && H5_VERS_MINOR <= 8 && (H5_VERS_MINOR < 8 || H5_VERS_RELEASE < 9)
#define HDF5_SUPPORTS_MEMORY_FILE
#endif
```
at the beginning of the file and use that macro after